widget/infprogressbar_test.go

```diff
@@ -0,0 +1,54 @@
+package widget
+
+import (
+	"testing"
+	"time"
+
+	"fyne.io/fyne"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInfiniteProgressBar_Creation(t *testing.T) {
+	bar := NewInfiniteProgressBar()
+	// ticker should be nil when created
+	assert.Nil(t, bar.ticker)
+}
+
+func TestInfiniteProgressBar_Ticker(t *testing.T) {
+	bar := NewInfiniteProgressBar()
+
+	bar.Show()
+	// Show() starts a goroutine, so pause for it to initialize
+	time.Sleep(10 * time.Millisecond)
+	assert.NotNil(t, bar.ticker)
+	bar.Hide()
+	assert.Nil(t, bar.ticker)
+
+	// make sure it restarts when re-shown
+	bar.Show()
+	// Show() starts a goroutine, so pause for it to initialize
+	time.Sleep(10 * time.Millisecond)
+	assert.NotNil(t, bar.ticker)
+	bar.Hide()
+	assert.Nil(t, bar.ticker)
+}
+
+func TestInfiniteProgressRenderer_Layout(t *testing.T) {
+	bar := NewInfiniteProgressBar()
+	width := 100
+	bar.Resize(fyne.NewSize(width, 10))
+
+	render := Renderer(bar).(*infProgressRenderer)
+
+	// width of bar is 1/50 of width after initial creation
+	assert.Equal(t, width/50, render.bar.Size().Width)
```

Comment:
I would highly recommend moving minBarWidth and maxBarWidth (with appropriate naming to avoid conflicts) to package consts so that they can be referenced in this test.
Magic numbers like this should be avoided if possible.

Reply:
This was a good suggestion, cleaned up the tests some too :)